pkg/webui/console/store/actions/gateway.js

```diff
@@ -32,52 +25,70 @@ import {
   createStopEventsStreamActionType,
   clearEvents,
   createClearEventsActionType,
-} from '../actions/events'
+} from './events'
 
-import {
-  getApiKey,
-  createGetApiKeyActionType,
-  getApiKeySuccess,
-  createGetApiKeySuccessActionType,
-  getApiKeyFailure,
-  createGetApiKeyFailureActionType,
-} from './api-key'
+import { createGetApiKeyActionType } from './api-key'
+
+import { createRequestActions } from './lib'
 
 export const SHARED_NAME = 'GATEWAY'
 
-export const GET_GTW = 'GET_GATEWAY'
-export const GET_GTW_SUCCESS = 'GET_GATEWAY_SUCCESS'
-export const GET_GTW_FAILURE = 'GET_GATEWAY_FAILURE'
+export const GET_GTW_BASE = 'GET_GATEWAY'
+export const [{
+  request: GET_GTW,
+  success: GET_GTW_SUCCESS,
+  failure: GET_GTW_FAILURE,
+}, {
+  request: getGateway,
+  success: getGatewaySuccess,
+  failure: getGatewayFailure,
+}] = createRequestActions(GET_GTW_BASE,
+  id => ({ id }),
+  (id, selector) => ({ selector })
+)
+
+export const UPDATE_GTW_STATS_BASE = 'UPDATE_GATEWAY_STATISTICS'
+export const [{
+  request: UPDATE_GTW_STATS,
+  success: UPDATE_GTW_STATS_SUCCESS,
+  failure: UPDATE_GTW_STATS_FAILURE,
+}, {
+  request: updateGatewayStatistics,
+  success: updateGatewayStatisticsSuccess,
+  failure: updateGatewayStatisticsFailure,
+}] = createRequestActions(UPDATE_GTW_STATS_BASE, id => ({ id }))
+
+export const GET_GTW_API_KEY_BASE = createGetApiKeyActionType(SHARED_NAME)
+export const [{
+  request: GET_GTW_API_KEY,
+  success: GET_GTW_API_KEY_SUCCESS,
+  failure: GET_GTW_API_KEY_FAILURE,
+}, {
+  request: getGatewayApiKey,
+  success: getGatewayApiKeySuccess,
+  failure: getGatewayApiKeyFailure,
+}] = createRequestActions(GET_GTW_API_KEY_BASE, (gtwId, keyId) => ({ entityId: gtwId, keyId }))
```

Comment:
Note that you are not using predefined helpers from `actions/api-key.js`, same for `actions/api-keys.js` that can raise inconsistencies in parameters across different action creators for apps/gtws.

I would prefer something like this:
```
const createGetApiKeyBaseActionType = name => (
  `GET_${name}_API_KEY`
)

export default name => createRequestActions(
  createGetApiKeyActionType(name),
  // payload creator,
  // meta creator
)
```

otherwise delete unused exports in `actions/api-key.js`, `actions/api-key.js`